pygfx/utils/load.py

```diff
@@ -5,72 +5,432 @@
+          - False: ignore any camera in the file but add a camera to scene
+          - None: no camera is imported
+    background : bool
+        Trimesh scenes typically have a white background. If True, we will add a white background
+        to the pygfx scene. If False, no background will be added.
+
+    Returns
+    -------
+    scene : pygfx.Scene
+        The scene converted to pygfx.
+
+    """
+    import trimesh  # noqa
+
+    # Convet single meshes into a scene (this makes the code below much simpler)
+    if isinstance(tm_scene, trimesh.Trimesh):
+        tm_scene = trimesh.Scene(geometry=tm_scene)
+    elif not isinstance(tm_scene, trimesh.Scene):
+        raise ValueError(f"Unexpected trimesh data: {type(tm_scene)}")
+
+    # Basic scene setup
+    gfx_scene = gfx.Scene()
+
+    # Take care of meshes
+    # Note: we're traversing the scene graph only for meshes (i.e. not lights, cameras, etc.)
+    # That's because as far as I can tell, trimesh's scene graph only applies to geometries
+    # while lights and cameras are stored separate from 'world'.
+    if meshes:
+        if not flatten:
+            # By convention, we expect the "world" objects to be the root of the scene graph
+            G = tm_scene.graph.to_networkx()
+            if "world" not in G.nodes:
+                raise ValueError("No 'world' node found in scene graph")
+
+            # Load the geometries and materials
+            gfx_geometries, gfx_materials = objects_from_trimesh(
+                tm_scene, materials=materials
+            )
+
+            def _build_graph_bfs(node, node_group):
+                """Recursively parse scene graph into pygfx.Groups."""
+                # Note: not sure if that will ever happen in practice but
+                # in theory, this implementation could run into the recursion
+                # depth limit (3000 on my machine). If that turns out to be
+                # a problem we need to switch the implementation
+
+                # Go over this node's children
+                for child in tm_scene.graph.transforms.children.get(node, []):
+                    # Is this a leaf node?
+                    is_leaf = G.out_degree[child] == 0
+
+                    # Does it have a geometry?
+                    geometry_name = G.edges[(node, child)].get("geometry", None)
+                    # See if this child has a geometry
+                    if geometry_name is not None:
+                        # Create the geometry
+                        mesh = gfx.Mesh(
+                            gfx_geometries[geometry_name],
+                            gfx_materials[geometry_name],
+                        )
+                    else:
+                        mesh = None
+
+                    # If this is not a leaf node, we need to use a Group
+                    if not is_leaf:
+                        # Generate a new graph for this child
+                        child_group = gfx.Group()
```

Comment:
Just to be clear, you're not technically required to use a Group, all WorldObjects can have children. So if the current node has geometry, you could just create a Mesh and add children to it.

Not sure if that would complicate the code unnecessarily, but it would help to keep the node count low, which is beneficial for performance.

Reply:
Ah, good point! Originally I thought, I'd stick as close as possible to the trimesh scene graph in case the use of groups is intentional. However, it seems that in the trimesh scene graph WorldObjects can't have children directly, i.e. they _have_ to use groups.  

With f56bcf7 this is now only using `gfx.Groups` when absolutely necessary.